embassy-usb-logger/src/lib.rs

```diff
@@ -105,6 +105,34 @@ impl<const N: usize> UsbLogger<N> {
             join(run_fut, join(log_fut, discard_fut)).await;
         }
     }
+
+    // Creates the futures needed for the logger from a given class
+    pub async fn create_future_from_class<'d, D>(&'d self, class: CdcAcmClass<'d, D> )
```

Comment:
I tried to use `create_future_from_class` inside `run` to not duplicate the code, but couldn't make it work since neither the class nor sender and receiver can be copied. If somebody finds a good way to clean this up, feel free to let me know